components/places/src/storage/bookmarks.rs

```diff
@@ -426,6 +431,110 @@ impl Default for UpdateTreeLocation {
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct ReadonlyBookmark {
+    pub guid: SyncGuid,
+    pub parent_guid: Option<SyncGuid>,
+    // Always 0 if parent_guid is None
+    pub position: BookmarkPosition,
+    pub date_added: Timestamp,
+    pub last_modified: Timestamp,
+    pub url: Option<Url>,
+    pub title: Option<String>,
+}
+
+impl From<ReadonlyBookmark> for ReadonlyItem {
+    fn from(b: ReadonlyBookmark) -> Self {
+        ReadonlyItem::Bookmark { b }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct ReadonlySeparator {
+    pub guid: SyncGuid,
+    pub date_added: Timestamp,
+    pub last_modified: Timestamp,
+    pub parent_guid: Option<SyncGuid>,
+    pub position: BookmarkPosition,
+}
+
+impl From<ReadonlySeparator> for ReadonlyItem {
+    fn from(s: ReadonlySeparator) -> Self {
+        ReadonlyItem::Separator { s }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct ReadonlyFolder {
+    pub guid: SyncGuid,
+    pub date_added: Timestamp,
+    pub last_modified: Timestamp,
+    pub parent_guid: Option<SyncGuid>,
+    pub position: BookmarkPosition,
+    pub title: Option<String>,
+    pub child_guids: Option<Vec<SyncGuid>>,
+    pub child_nodes: Option<Vec<ReadonlyItem>>,
+}
+
+impl From<ReadonlyFolder> for ReadonlyItem {
+    fn from(f: ReadonlyFolder) -> Self {
+        ReadonlyItem::Folder { f }
+    }
+}
+
+// The type used to update the actual item.
+#[derive(Debug, Clone)]
+pub enum ReadonlyItem {
+    Bookmark { b: ReadonlyBookmark },
+    Separator { s: ReadonlySeparator },
+    Folder { f: ReadonlyFolder },
+}
+
```

Comment:
I think we should get a ticket on file to kill `PublicNode` - I don't think it's offering any value to use at the moment.

Reply:
Very much agree--I created a ticket in the after places uniffication epic for this.